ch_tools/monrun_checks/ch_backup.py

```diff
@@ -114,35 +147,34 @@ def check_backup_age(ch_client, backups, *, age_threshold=timedelta(days=1), cri
         die(2, "Didn't find a backup to check")
 
     backup_age = get_backup_age(checking_backup)
-    if backup_age < age_threshold:
+
+    if crit_threshold and backup_age >= crit_threshold:
+        status = 2
+    elif warn_threshold and backup_age >= warn_threshold:
+        status = 1
+    else:
         return
 
     if checking_backup["state"] == "creating":
         message = f"Last backup was started {backup_age.days} days ago"
     else:
         message = f"Last backup was created {backup_age.days} days ago"
-    status = 1
-    if crit is not None and uptime >= crit and backup_age >= crit:
-        status = 2
+
     die(status, message)
 
 
-def check_backup_count(config: BackupConfig, backups: List[Dict]) -> None:
+def check_backup_count(backups: List[Dict], backup_count_warn_threshold: int) -> None:
     """
     Check that the number of backups is not too large.
     """
-    max_count = (
-        config.retain_count + config.deduplication_age_limit.days + 2
-    )  # backup-service backup retention delay
+    backup_count = len(backups)
+    if backup_count_warn_threshold and backup_count >= backup_count_warn_threshold:
+        die(
+            1, f"Too many backups exist: {backup_count} > {backup_count_warn_threshold}"
+        )
 
-    count = sum(
-        1 for backup in backups if backup.get("labels", {}).get("user_initiator")
-    )
-    if count > max_count:
-        die(1, f"Too many backups exist: {count} > {max_count}")
```

Comment:
Complex logic that produces a lot of false warnings in prod.

Reply:
I haven't seen a lot of this alerts in prod, but believe that there were a lot. But now we can get a lot of permanent warning in prod for clusters with a lot of user created backups. But maybe it's better.

Reply:
https://paste.yandex-team.ru/fb0de08e-0c55-4684-9f0e-7888cff2656f/text

Notes:
- In most cases, warnings are not accounted in Juggler aggregates.
- Default value for `--backup-count-warn-threshold` is 0 - do not check backup count.